doc/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,370 @@
+##### 3.6.2.2.1
+Syntax
+
+Vrf and distance metric are optional in the CLI.
+
+```
+ip route vrf <vrf-name> <prefix: A.B.C.D/mask> {[interface <interface-name>] | [<next-hop-ip>] | [<next-hop-ip> interface <interface-name>]} <distance Metric>
+```
+
+```
+ipv6 route vrf <vrf-name> <prefix: A.B.C.D/mask> {[interface <interface-name>] | [<next-hop-ip>] | [<next-hop-ip> interface <interface-name>]} <distance Metric>
+
+```
+
+
+Syntax Description:
+
+|    Keyword    | Description |
+|:-----------------:|:-----------:|
+| vrf | The static routes will be configured in the specified VRF. If not specified, default VRF will be used.
+| prefix | This is the destination prefix for IPv4 or IPv6 routes.
+| interface | Specifies the interface to be used for this route. If the interface doesn't belong to the current VRF, it is considered a route-leak. The VRF to which this interface belongs will be set as the nexthop-vrf in the STATIC_ROUTE table .
+| next-hop-ip | This provides the gateway IP for the prefix.
+| distance Metric| Specifies distance value for this route. Value form 1 to 255.
+
+
+Example:
+
+#### IPv4 examples
+````
+sonic(config)# ip route 10.1.1.1/24 10.1.1.3
+
+In configDB new STATIC_ROUTE table will be filled with following entries for default VRF and prefix:-
+key                 = STATIC_ROUTE|default|10.1.1.1/24;
+vrf-name            = default
+prefix              = 10.1.1.1/24
+nexthop             = 10.1.1.3
+ifname              = ""
+distance            = 0
+nexthop-vrf         = ""
+
+
+sonic(config)# ip route 10.1.1.1/24 Ethernet12 20
+
+Assumption is Ethernet12 is configured in Vrf-RED by user.
+STATIC_ROUTE table entries are updated for newly added route:-
+key                 = STATIC_ROUTE|default|10.1.1.1/24;
+vrf-name            = default
+prefix              = 10.1.1.1/24
+nexthop             = 10.1.1.3, 0.0.0.0
+ifname              = "", Ethernet12
+distance            = 0, 20
+nexthop-vrf         = "", Vrf-RED.
+````
+
+````
+
+sonic(config)# ip route vrf Vrf-RED 10.5.6.6/24 Ethernet0 10
+
+Assumption is Ethernet0 is in default VRF.
+A new STATIC_ROUTE table will be filled with following entries for Vrf-RED and prefix:-
+key                 = STATIC_ROUTE|Vrf-RED|10.5.6.6/24;
+vrf-name            = Vrf-RED
+prefix              = 10.5.6.6/24
+nexthop             = 0.0.0.0
+ifname              = Ethernet0
+distance            = 10
+nexthop-vrf         = default
+
+
+````
+#### IPv6 examples
+````
+sonic(config)# ipv6 route 2::/64 Ethernet16
+
+sonic(config)# ipv6 route 2001:FF21:1:1::/64 18:2:1::1 100
+````
+##### 3.6.2.2.2 ipv4 /ipv6 Command to delete a next-hop.
+
+The vrf is an optional parameter.
+
+```
+no ip route vrf <vrf-name> <prefix: A.B.C.D/mask> {[interface <interface-name>] | [<next-hop-ip>] | [<next-hop-ip> interface <interface-name>]}
+
+```
+
+```
+no ipv6 route vrf <vrf-name> <prefix: A.B.C.D/mask> {[interface <interface-name>] | [<next-hop-ip>] | [<next-hop-ip> interface <interface-name>]}

```

Comment:
Need examples of delete routes. nexhop or interface is optional right

Reply:
It's mandatory, please refer to the above syntax.

Reply:
examples added